R/reportingengine-env.R

```diff
@@ -352,97 +228,21 @@ getStatisticsFromType <- function(statisticsType) {
-#'
-#' # Set the default legend caption displayed for range
-#' setDefaultTimeProfileStatistics(rangeCaption = "90% population range")
-#' }
-#'
-setDefaultTimeProfileStatistics <- function(statisticsType = NULL,
-                                            y = NULL,
-                                            ymin = NULL,
-                                            ymax = NULL,
-                                            yCaption = NULL,
-                                            rangeCaption = NULL) {
-  validateIsIncluded(statisticsType, StatisticsTypes, nullAllowed = TRUE)
-  # Allow user to enter the function directly
-  validateIsOfType(y, c("character", "closure"), nullAllowed = TRUE)
-  validateIsOfType(ymin, c("character", "closure"), nullAllowed = TRUE)
-  validateIsOfType(ymax, c("character", "closure"), nullAllowed = TRUE)
-
-  if (!isEmpty(statisticsType)) {
-    reEnv$defaultTimeProfileStatistics <- getStatisticsFromType(statisticsType)
-  }
-  # Assign variables to reEnv only if defined
-  eval(parseVariableToObject(
-    objectName = "reEnv$defaultTimeProfileStatistics",
-    variableName = c("y", "ymin", "ymax", "yCaption", "rangeCaption"),
-    keepIfNull = TRUE
-  ))
-
-  return(invisible())
-}
-
-
-#' @title getDefaultMCRandomSeed
-#' @description Get the default random seed when performing Monte Carlo sampling
-#' @return Random seed number
-#' @export
-#' @examples
-#' getDefaultMCRandomSeed()
-#'
-getDefaultMCRandomSeed <- function() {
-  return(reEnv$defaultMCRandomSeed)
-}
-
-#' @title setDefaultMCRandomSeed
-#' @description Set the default random seed when performing Monte Carlo sampling
-#' @param seed Random seed number as an integer
-#' @export
-#' @examples
-#' setDefaultMCRandomSeed(123456)
-#'
-setDefaultMCRandomSeed <- function(seed) {
-  validateIsInteger(seed)
-  reEnv$defaultMCRandomSeed <- seed
-  return(invisible())
-}
-
-#' @title getDefaultMCRepetitions
-#' @description Get the default number of repetitions when performing Monte Carlo sampling
-#' @return Random seed number
+#' @title reSettingsNames
+#' @description
+#' Names of the default/global settings stored in `reEnv`.
+#' Can be used with `getRESettings()`
+#' @import ospsuite.utils
 #' @export
-#' @examples
-#' getDefaultMCRepetitions()
-#'
-getDefaultMCRepetitions <- function() {
-  return(reEnv$defaultMCRepetitions)
-}
+#' @family enum helpers
+reSettingsNames <- enum(names(reEnv))
 
-#' @title setDefaultMCRepetitions
-#' @description Set the default number of repetitions when performing Monte Carlo sampling
-#' @param n Number of repetitions
-#' @export
-#' @examples
-#' setDefaultMCRepetitions(1e4)
-#'
-setDefaultMCRepetitions <- function(n) {
-  validateIsInteger(n)
-  reEnv$defaultMCRepetitions <- n
-  return(invisible())
-}
+# Save a sync version of re-env as an .RData file
+newEnv <- reEnv
+save(
+  "newEnv",
+  file = file.path(system.file(package = "ospsuite.reportingengine"), "extdata", "re-env.RData")
+)
```

Comment:
When is this called?
We should assume that the package installation location is read-only and don't write anything there during runtime.

Reply:
It is called when I update the documentation of the package: `devtools::document()` and not exported/exposed to the user.